### ISSUE TITLE
default charset to utf-8 when open csv file by application (MS Office)

### ISSUE THREAD

**Issue by thucnd** [nrinaudo/kantan.csv#69]
KantaCsv write data to csv file with format UTF-8 But When I open file in MS Excel, it changed to shift_jis. 
Shift_jis cannot display some special characters in Japanese ( “－"　“～”)

Solution:
Csv files starts with BOM (utf-8 encoding), Excel opened correctly. 

```
def write(data: Seq[R], name: String): Try[File] = {
    Try {
      val out = java.io.File.createTempFile(name, ".csv")
      val writer = new OutputStreamWriter(new FileOutputStream(out), "utf-8")
      writer.write('\uFEFF')
      writer.asCsvWriter[R](',', header: _*).write(data)
      writer.close()
      out
    }.recover {
      case e: Exception => throw new FailCsvWriting("error.csv.failWriting")
    }
  }
```

**Comment by nrinaudo** [comment]
Agreed. Working on a more permanent fix - the current target implementation is to explicitly import a `bom` package which will override the way we turn byte-streams into character streams to add BOM support:
- add a BOM when writing textual data in an encoding that has one.
- attempt to find a BOM when reading text and, if one is found, use the corresponding encoding (potentially ignoring the one specified by the developer).

These behaviours are not going to be enabled by default as BOMs are very Microsoft-specific and certainly not part of the official CSV specifications.

**Comment by nrinaudo** [comment]
Right, so this is implemented in `0.1.17-SNAPSHOT`, available on the sonatype repository. In order to retrieve this, you'll need to make sure the following in your SBT resolvers:

```scala
resolvers := Seq(Resolver.sonatypeRepo("snapshots"))
```

Once that's done, just import `kantan.codecs.resource.bom._`, and you should be able to rewrite your code as:

```scala
def write(data: Seq[R], name: String): Try[File] =
  Try { java.io.File.createTempFile(name, ".csv").writeCsv(data, ',', header:_*) }
    .recover {
      case _: Exception => throw new FailCsvWriting("error.csv.failWriting")
    }
```

**Comment by nrinaudo** [comment]
I'll consider this closed for now - my fix seems to work, at least for me - but feel free to re-open if you feel something is  still missing.
